ProjectVoyage/View Controllers/Chat message/MessagesViewController.m

```diff
@@ -366,21 +386,24 @@ - (UIView *)tableView:(UITableView *)tableView viewForHeaderInSection:(NSInteger
     return view;
 }
 
+- (void) scrollToBottomAnimated:(BOOL)animated {
+    // CC-  This if statement was in place to check the case where you opened a chat
+    // and there were more loaded messages than it could be possibly be visible on screen
+    // if (_chat.messages.count > [[_tableView visibleCells] count]) {
+    NSIndexPath *indexPath = [NSIndexPath indexPathForRow:(_chat.messages.count - 1) inSection:0];
+    [_tableView scrollToRowAtIndexPath:indexPath atScrollPosition:UITableViewScrollPositionBottom animated:animated];
+}
 
-- (void)tableViewScrollToBottomAnimated:(BOOL)animated {
-    NSInteger numberOfRows = self.chat.messages.count;
-    
-    // Open chat in the very last message
-    if (numberOfRows) {
-        NSIndexPath *indexPath = [NSIndexPath indexPathForRow:(self.chat.messages.count - 1) inSection:0];
-        
-        [_tableView scrollToRowAtIndexPath:indexPath
-                    atScrollPosition:UITableViewScrollPositionBottom
-                    animated:animated];
+- (void)scrollViewDidScroll:(UIScrollView *)scrollView {
+    if (_shouldKeepScrolling) {
+        if (scrollView.contentOffset.y + _tableView.safeAreaInsets.top == 0) {
```

Comment:
nit: save the sum as a variable just to better explain what this check is doing?